### PR TITLE
Improve scheduler driver logging to always include exit code, stdout, stderr

### LINKED ISSUE #7691: Fix LSF kill logging when matching the stdout
In LSF driver we match `stdout` when the `returncode==0` and we should perhaps do it when `returcode!=0` and we should add filtering of `stderr` also. This should include filtering of the message: 
`Job has already finished`

### PR DESCRIPTION
**Pull request by jonathan-eq**
**Issue**
Resolves #7691 

**Approach**
^

(Screenshot of new behavior in GUI if applicable)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/scheduler/driver.py [hunk 1]
```diff
@@ -84,25 +84,29 @@ async def _execute_with_retry(
...
 
             assert process.returncode is not None
+            outputs = (
+                f"exit code {process.returncode}, "
+                f'output: "{stdout.decode(errors="ignore").strip() or "<empty>"}", and '
+                f'error: "{stderr.decode(errors="ignore").strip() or "<empty>"}"'
+            )
             if process.returncode == 0:
-                return True, stdout.decode(errors="ignore").strip()
+                return True, outputs
```
Comment (xjules) [inline src/ert/scheduler/driver.py:93]: Apparently this changes behavior of some tests.

Comment (jonathan-eq) [reply src/ert/scheduler/driver.py:93]: Yes, I will fix those tests, but I am not sure it is a good idea to test the logging format..

File: src/ert/scheduler/driver.py [hunk 2]
```diff
@@ -84,25 +84,35 @@ async def _execute_with_retry(
             stdout, stderr = await process.communicate(stdin)
 
             assert process.returncode is not None
+            outputs = (
+                f"exit code {process.returncode}, "
+                f'output: "{stdout.decode(errors="ignore").strip() or "<empty>"}", and '
```
Comment (xjules) [inline src/ert/scheduler/driver.py:89]: why this weird formatting? I'd change make `" "` the outer ones. Also do we need `" "` inside the error message?

Comment (berland) [reply src/ert/scheduler/driver.py:89]: I think I prefer including quotes (single or double) to wrap the stdout/stderr output to distinguish it from the log from our code.

**Review by xjules — APPROVED** [review]
I think it can go now! You need to fix some conflicts though. :rocket:

**Pull request merged**